base/src/main/java/io/spine/base/FieldPaths.java

```diff
@@ -59,25 +61,46 @@ private FieldPaths() {
     /**
      * Parses the given field path into a {@link FieldPath}.
      *
-     * @param stringPath
+     * @param fieldPath
      *         non-empty field path
      * @return parsed field path
+     * @deprecated please use {@link Field#parse(String)}
      */
-    public static FieldPath parse(String stringPath) {
-        checkNotNull(stringPath);
-        checkArgument(!stringPath.isEmpty(), "Path must not be empty.");
+    @Deprecated
+    public static FieldPath parse(String fieldPath) {
+        checkNotNull(fieldPath);
+        return doParse(fieldPath);
+    }
 
-        List<String> pathElements = dotSplitter.splitToList(stringPath);
-        return fromElements(pathElements);
+    static FieldPath doParse(String fieldPath) {
```

Comment:
Instead of adding `doSmth` methods, let's move the logic into the `Field` altogether. This way, when the time comes, we can just delete the `FieldPaths.java` without any further migration.